### DISCUSSION TITLE
Issue Compiling Lib Following Guide for Zephyr

### DISCUSSION THREAD

**Discussion by MaxMaeder** [pschatzmann/arduino-audio-tools#1319] [Q&A]
Hello again,

I followed the guide to use the framework without Arduino, but unfortunately am getting some strange compilation errors when I compile for an nrf91 on the Zephyr platform. Any help would be really appreciated!

Thanks,
Max
[error.log](https://github.com/pschatzmann/arduino-audio-tools/files/14234068/error.log)
[CMakeLists.txt](https://github.com/pschatzmann/arduino-audio-tools/files/14234069/CMakeLists.txt)
[prj.conf.txt](https://github.com/pschatzmann/arduino-audio-tools/files/14234070/prj.conf.txt)

**Comment by pschatzmann** [comment]
This is strange: My gut feeling is that there might be some missing includes. In order to address the first error I suggest to add #include "AudioTools/BaseConverter.h" to AudioStreams.h.

 But I would not have any explanation why this error does not show up in Arduino or on the desktop builds...

**Reply by MaxMaeder** [reply]
Hmm, that actually fixed most of the errors, thanks. Only ones I'm getting now are:
```
In file included from .././arduino-audio-tools/src/AudioTools/AudioStreams.h:7,
                 from ../src/main.cpp:17:
.././arduino-audio-tools/src/AudioEffects/SoundGenerator.h: In member function 'virtual T audio_tools::SineWaveGenerator<T>::readSample()':
.././arduino-audio-tools/src/AudioEffects/SoundGenerator.h:201:30: error: there are no arguments to 'sinf' that depend on a template parameter, so a declaration of 'sinf' must be available [-fpermissive]
  201 |     T result = m_amplitude * sinf(angle);
      |                              ^~~~
.././arduino-audio-tools/src/AudioEffects/SoundGenerator.h:201:30: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
.././arduino-audio-tools/src/AudioEffects/SoundGenerator.h: In member function 'int audio_tools::GeneratorFromArray<T>::setupSine(int, float, float)':
.././arduino-audio-tools/src/AudioEffects/SoundGenerator.h:546:18: error: there are no arguments to 'sin' that depend on a template parameter, so a declaration of 'sin' must be available [-fpermissive]
  546 |       table[j] = sin(j * angle) * amplitude;
      |                  ^~~
[254/263] Linking C static library zephyr/kernel/libkernel.a
```

**Reply by MaxMaeder** [reply]
If I comment out those sin functions, it does compile, strange. It would be nice if the `SoundGenerator.h` class worked though, for testing sinks.

**Reply by pschatzmann** [reply]
It is definitely not good that different sin() functions are used. I tend to prefer the C sinf() implementation. 
Maybe this is also an issue with an include.

Double check how to use sinf in your environment...

**Reply by pschatzmann** [reply]
Maybe adding #include <math.h> will do the trick...

**Reply by MaxMaeder** [reply]
Thanks so much, that's fixed it. Should I make a PR with those changes for `#ifndef ARDUINO`?

**Reply by pschatzmann** [reply]
No, I just committed the corrections
